meson.build

```diff
@@ -0,0 +1,182 @@
+project(
+  'cloud-init',
+  meson_version: '>=0.63.0', # rockylinux/9
+  license: 'GPL-3 or Apache-2.0',
+  default_options: [
+    # The default can yield broken results.
+    'python.install_env=auto',
+  ],
+)
+system = host_machine.system()
+sysconfdir = get_option('sysconfdir')
+INIT_SYSTEM = get_option('init_system')
+
+lib_exec_dir = join_paths('/usr', get_option('libexecdir'), 'cloud-init')
+
+pymod = import('python')
+python = pymod.find_installation('python3')
+
+#https://mesonbuild.com/FAQ.html#but-i-really-want-to-use-wildcards for downsteam drop-in files
+find = find_program('find')
+
+install_subdir(
+  'cloudinit',
+  install_dir: python.get_install_dir(),
+  install_tag: 'python',
+)
+
+completions = dependency('bash-completion')
+# Binaries and script entrypoints
+bash_completions_dir = completions.get_variable(
+  pkgconfig: 'completionsdir',
+  default_value: join_paths(get_option('datadir'), 'bash-completion', 'completions'),
+)
+completions = dependency('bash-completion')
+install_data(
+  'bash_completion/cloud-init',
+  install_dir: bash_completions_dir,
+  install_tag: 'scripts',
+)
+
+install_data(
+  [
+    'tools/ds-identify',
+    'tools/hook-hotplug',
+    'tools/uncloud-init',
+    'tools/write-ssh-key-fingerprints',
+  ],
+  install_dir: lib_exec_dir,
+  install_tag: 'scripts',
+)
+
+install_data(
+  ['tools/cloud-init-per', 'scripts/cloud-id', 'scripts/cloud-init'],
+  install_mode: 'rwxr-xr-x',
+  install_dir: get_option('bindir'),
+)
+
+# Required Config and Templates
+render_tmpl = './tools/render-template'
+
+cfgs = run_command(find, 'config/cloud.cfg.d', '-type', 'f', '-name', '*.cfg', check: true)
+install_data(
+  ['config/cloud.cfg.d/README'] + cfgs.stdout().strip().split('\n'),
+  install_dir: join_paths(sysconfdir, 'cloud', 'cloud.cfg.d'),
+  install_tag: 'config',
+)
+
+templates = run_command(find, 'templates', '-type', 'f', '-name', '*.tmpl', check: true)
+install_data(
+  templates.stdout().strip().split('\n'),
+  install_dir: join_paths(sysconfdir, 'cloud', 'templates'),
+  install_tag: 'config',
+)
+
+if INIT_SYSTEM == 'systemd'
+  systemd = dependency('systemd')
+  udev = dependency('udev')
+  systemd_unit_dir = systemd.get_variable(pkgconfig: 'systemdsystemunitdir')
+  systemd_generator_dir = systemd.get_variable(pkgconfig: 'systemdsystemgeneratordir')
+  udev_dir = udev.get_variable(pkgconfig: 'udevdir', default_value: '/usr/lib/udev')
```

Comment:
I don't think that we should hard code absolute paths. Doing so defeats the flexibility of the build system and the ability for downstreams to override this when needed.

The libdir option might make sense here, but this also just shouldn't be necessary, right? If missing pkgconfig, this build will fail in other ways, and if pkgconfig cannot find the correct directory, that indicates a downstream packaging issue.